src/nodes.coffee

```diff
@@ -1894,6 +1897,48 @@ exports.Obj = class Obj extends Base
       answer.push @makeCode join
     if @front then @wrapInParentheses answer else answer
 
+  expandProperty: (property) ->
+    {variable, context, operatorToken} = property
+    key = if property instanceof Assign and context is 'object'
```

Comment:
This has some duplication with `Obj::compileNode()` but I don't see an easy way to share code (however, see comments re a missing `ObjectProperty` abstraction, which might change that)